AsyncEnumerator.cs

```diff
@@ -71,6 +72,8 @@ internal void SetCanceled()
                 _isCanceled = true;
                 _enumerationException = null;
                 Interlocked.Exchange(ref _completionLock, 0);
+                if (_resumeTCS != null && !_resumeTCS.Task.IsCompleted)
+                    _resumeTCS.SetException(new AsyncEnumerationDisposedException());
```

Comment:
But the exception type is not correct, because you don't known if the enumerator is actually disposed. See my comments to the `AsyncEnumerationDisposedException` type.